libpromises/evalfunction.c

```diff
@@ -650,14 +654,35 @@ static FnCallResult FnCallRandomInt(ARG_UNUSED EvalContext *ctx, ARG_UNUSED cons
         return FnFailure();
     }
 
+    if (to == from)
+    {
+        return FnReturnF("%d", from);
+    }
+
     if (from > to)
     {
         tmp = to;
         to = from;
         from = tmp;
     }
 
-    range = abs(to - from);
+    range = to - from;
+    assert(range > 0);
+
+    // Seed was specified:
+    if (finalargs->next->next != NULL)
+    {
+        const char *seed = RlistScalarValue(finalargs->next->next);
+
+        // Use beginning of SHA checksum as pseudo random number:
+        unsigned char digest[EVP_MAX_MD_SIZE + 1];
+        HashString(seed, strlen(seed), digest, HASH_METHOD_SHA256);
+        unsigned long converted_sha = ((unsigned long*)(digest))[0];
```

Comment:
Here you can use floating point division to avoid the modulo bias.

Reply:
And another good idea would be to use `uint64_t` and `UINT64_MAX` for the double division.